[PyTorch] Extract parseOperator() into a standalone source file

larryliu0820 opened the pull request:
Summary: This is following up this PR: https://github.com/pytorch/pytorch/pull/61862. The purpose is to modularize operator parsing so that it can be used as needed without pulling the whole `import.cpp` into build.

Test Plan: Added a unit test in `test_lite_predictor.cpp` called `ParseOperators`, similar to `ParseBytecode`.

Differential Revision: D31006555

iseeyuan commented on test/cpp/jit/test_lite_interpreter.cpp:
```diff
@@ -956,6 +957,59 @@ TEST(RunTimeTest, ParseBytecode) {
+  std::vector<IValue> constants{
+      to_tuple({1}),
+  };
+  int64_t model_version = 0x3L;
```
```suggestion
  int64_t model_version = kProducedBytecodeVersion;
```

iseeyuan commented on test/cpp/jit/test_lite_interpreter.cpp:
```diff
@@ -956,6 +957,59 @@ TEST(RunTimeTest, ParseBytecode) {
+  parseInstructions(
+      function_name, instructions, debug_handles_m_tuple, function.get());
+  parseOperators(operators, model_version, 1, function.get());
+  parseTypes(types, function.get());
```
This line may not be necessary

iseeyuan reviewed (approved):
LGTM. A couple of inline comments. The version 3 is going to be deprecated, please use the current version number.

larryliu0820 committed:
[PyTorch] Extract parseOperator() into a standalone source file (#65179)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/65179

This is following up this PR: https://github.com/pytorch/pytorch/pull/61862. The purpose is to modularize operator parsing so that it can be used as needed without pulling the whole `import.cpp` into build.

Test Plan: Added a unit test in `test_lite_predictor.cpp` called `ParseOperators`, similar to `ParseBytecode`.

Reviewed By: iseeyuan

Differential Revision: D31006555

fbshipit-source-id: 6a6980418ea9f75a7f107e8eea5812526e44ca43
